File: RemoveElement.py
class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        # the oj will truc list to be the length
        while elem in A:
            A.remove(elem)
        return len(A)

# if use for x in A, then there will be some error
A=[3,3]
elem=3
print(Solution().removeElement(A,elem))

File: cpp/线性表/数组/Remove Element.cpp
#include<algorithm>
using namespace std;
class Solution {
  public:
    int removeElement(int A[], int n, int elem) {
        return distance(A, remove(A, A + n, elem));
    }
};

/*class Solution {
  public:
    int removeElement(int A[], int n, int elem) {
        int index = 0; 
        for (int i = 0; i < n; ++i) {
            if (A[i] != elem) 
                A[index++] = A[i];
        }
        return index;
    }
};*/
